server/api/endpoints/inventory.js

```diff
@@ -0,0 +1,40 @@
+import { Inventory } from "../../../lib/collections/";
+
+export function inventory(Api) {
+  Api.addCollection(Inventory);
+  Api.addRoute("inventory", { authRequired: false }, {
+    get: () => {
+      return Inventory.find().fetch();
+    },
+    post: {
+      roleRequired: ["author", "admin"],
+      action: () => {
+        if (inventory.insert(this.request.body)) {
+          return { status: "success", message: "Inventory updated" };
+        }
+        return {
+          status: "fail",
+          message: "Update failed"
+        };
+      }
+    }
+  });
+
+  Api.addRoute("inventory/:id", { authRequired: false }, {
+    get: () => {
+      return inventory.findOne().fetch();
```

Comment:
.find({_id: this.urlParams.id})
.find({_id: ObjectId(this.urlParams.id)})

or check if `this.urlParams.id` has a value in the first place.